File: posts/mathworks-interview/49.py
from typing import *

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = dict()

        for v in strs:
            root = "".join(sorted(v))
            if root in res:
                res[root].append(v)
            else:
                res[root] = [v]

        return list(res.values())

File: posts/mathworks-interview/49.rs
struct Solution;

use std::collections::HashMap;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut res: HashMap<String, Vec<String>> = HashMap::new();

        for v in strs {
            let chars: Vec<char> = v.chars().collect();
            chars.sort();
            let root: String = chars.join("");

            match res.get(&root) {
                Some(array) => {
                    array.push(v.clone());
                },
                None => {
                    res.insert(root, vec![v.clone()]);
                }
            }
        }

        return res.values().cloned().collect::<Vec<String>>();
    }
}